### PR TITLE
[DDW-170] Password input with strength meter

### PR DESCRIPTION
**Pull request by DominikGuzei**
This PR integrates the new password input component from react-polymorph with built-in strength meter:

## Todos

- [x] Check and fix tests
- [x] Check all tooltips in the app for broken styles since i had to change the existing overrides to make the password input component look correctly
- [x] Check forms layout / design & tooltip overlays etc. for issues

## Screenshots

CleanShot 2020-06-18 at 16 56 57

CleanShot 2020-06-18 at 16 57 38

CleanShot 2020-06-18 at 16 58 05

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test

---

### Basics

- [x] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [x] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [x] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by nikolaglumac** [comment]
@darko-mijic we need text copy for this PR 🙏

**Comment by gnpf** [comment]
@DominikGuzei The password strength requirements are not implemented, so I can insert an insecure password that has at least 10 characters and it would be valid:

_Note: for this record, I changed the type of the input to "text" so the password can be visible:_

1FeC5Cjt14

**Comment by a-rukin** [comment]
image
for dark themes color of progress bg is wrong

**Comment by DominikGuzei** [comment]
@gnpf the only requirement is 10 characters from what i know cc @darko-mijic @nikolaglumac

**Comment by DominikGuzei** [comment]
The tooltips on the connecting/loading screen are broken, im working on it

**Comment by a-rukin** [comment]
image

is this broken by this PR?

**Comment by DominikGuzei** [comment]
@a-rukin see my comment right before yours 😅 yes, im fixing this atm

**Comment by DominikGuzei** [comment]
@a-rukin loading screen tooltips should be fixed now 👍

**Comment by gnpf** [comment]
@DominikGuzei The strength meter and tooltips are displayed on the "Repeat" field if the password was not inserted first on the "insert" field. Is this expected?

5ynuidyWTt

**Comment by DominikGuzei** [comment]
@gnpf good catch, this is a bug - working on it now 👍

**Comment by DominikGuzei** [comment]
@gnpf the password repeat issue should be fixed 👍 
PS: Please also double-check the fix for the transactions list, since I had to solve some weird merge conflicts

**Comment by ManusMcCole** [comment]
Probly not much of an issue at all just a minor improvement. When changing password and the user enters incorrect password, then changes the new password input fields and deletes all of the characters the red highlighter around the current password field does not come back. Maybe there is no time just a suggestion 
redboxmisplacement

**Comment by gnpf** [comment]
> @gnpf the password repeat issue should be fixed 👍
> PS: Please also double-check the fix for the transactions list, since I had to solve some weird merge conflicts

@DominikGuzei The fix looks good and I'm still checking on the transactions tab. Anyway, I found a little issue and I'm wondering if it has something to do with those merge conflicts you mentioned, as it's doesn't reproduce on the last version of develop:

image

**Comment by DominikGuzei** [comment]
@ManusMcCole I looked into the issue: this is not related to this PR and happened the same way before. The reason is that the form actually does not validate the password itself but only if the user has entered anything at all. So, when you submit a wrong "current password" and then change the new password the following happens:

1. Invalid wallet password error is shown
2. you change the new password
3. local form validation is re-run -> the current password becomes "valid" from the perspective of the form, the error is removed.
4. you re-submit and get the error again (because the error comes from the API, not Daedalus)

Will create a ticket for this but don't think that it should be a blocker for this PR 👍

**Comment by DominikGuzei** [comment]
@gnpf yes, that looks like a result from the merge conflicts - I will re-merge develop into my branch to see if it goes away 👍

File: package.json [hunk 1]
```diff
@@ -178,7 +178,7 @@
     "csv-stringify": "5.3.6",
     "cucumber-html-reporter": "5.1.0",
     "electron": "8.2.2",
-    "electron-log-daedalus": "2.2.21",
+    "electron-log-daedalus": "2.2.20",
```
Comment (nikolaglumac) [inline package.json:181]: @DominikGuzei why did you downgrade this package?

**Comment by DominikGuzei** [comment]
@nikolaglumac final cleanup done 👍

**Pull request merged**